lib/Common/Memory/RecyclerWriteBarrierManager.cpp

```diff
@@ -66,6 +66,16 @@ X64WriteBarrierCardTableManager::OnThreadInit()
     ::GetCurrentThreadStackLimits(&stackEnd, &stackBase);
 #endif
 
+#ifdef X64_WB_DIAG
+    this->_stackbase = (char*)stackBase;
+    this->_stacklimit = (char*)stackEnd;
+#endif
+
+    // on Windows server 2012 stack limit can expand with process running, and causes
+    // accessing uncommitted card table page.
+    // TODO: use VirtualQuery twice to get the max possible stack limit
+    stackEnd -= AutoSystemInfo::PageSize * AutoSystemInfo::PageSize;
```

Comment:
the value is one page of card table, that represents 16MB address space, should be enough with stack growth. otherwise we can use virtual query to get the possible growth limit, which can be huge. 

---
In reply to: [90006505](https://github.com/Microsoft/ChakraCore/pull/2105#discussion_r90006505) [](ancestors = 90006505)